subsys/shell/shell_uart.c

```diff
@@ -274,6 +274,11 @@ static int enable_shell_uart(struct device *arg)
 		(CONFIG_SHELL_BACKEND_SERIAL_LOG_LEVEL > LOG_LEVEL_DBG) ?
 		CONFIG_LOG_MAX_LEVEL : CONFIG_SHELL_BACKEND_SERIAL_LOG_LEVEL;
 
+	if (dev == NULL) {
+		__ASSERT_NO_MSG(dev);
```

Comment:
having asserts and error checking in the same spot is not ideal IMO, you either assert on the whole thing or do error handling and propagation.

Reply:
Agreed. Fixed, thanks for review!